### DISCUSSION TITLE
[Feature] Force thumbnail regeneration and/or face detection on one picture

### DISCUSSION THREAD

**Discussion by chodthewacko** [immich-app/immich#8667] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

We could really use an option, when viewing a picture, to force thumbnail regeneration and/or face detection for that specific picture.  

When browsing my large library, I will occasionally hit a picture with a very clear face that isn't detected.
Even weirder, when the same picture is imported to another account, that faces was detected.
Sometimes all the faces are missing, sometimes it's just a handful of faces that are missing.
Either way, 'the refresh /redo missing' options in 'jobs' don't attempt to redo the face.

There should be an option to force a redo of the thumbnail and face detection for this individual picture.
The only way right now is to:
1)  Do a force redo of all faces, which will take ages for a large library and cause you to lose all your names.
2)  (maybe) modify the timestamp of a picture so that it gets picked up as a new picture and redone.
     This is not easy to do for internal libraries  (external libraries you can just 'touch')

Having this option would make it MUCH easier to figure out what face detection settings you need to use for particularly tricky albums.  We're talking literally minutes to test a set of settings versus hours (or days, in my case).

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
Thanks for the feedback! We're aware of the inflexibility here and plan to add a way to redo face detection for one or more images.

**Reply by 6543** [reply]
it would be nice if it would be next to the "rescan metadata" option:
![image](https://github.com/immich-app/immich/assets/24977596/eacd5ca0-15b3-4e73-842b-fbbb75f2605d)

witch is available if you select one or multiple images

**Comment by chodthewacko** [comment]
I've been playing around with this, and have found a workaround, of sorts.  This is on 1.100.0.
This is specifically for an external library, but you might be able to modify the process for internal library.

1)  Update the photo's modify timestamp.
    For the 'bad' photo, look at the 'info' to find the file name/location.  let's assume it's IMG_1111.jpg
    Update the file's modify timestamp using something a utility like 'touch'.

2)  In external libraries, click the 'three dots' on the right of that library and 're-scan all library files'.
    (It feels like you should be allowed to first set the library's scan settings to exclude !(**/IMG_1111.jpg),
     to make this run faster, but that doesn't seem to work at all).  Note this refreshes the thumbnails for that file, but not the faces.

3)  After the scan, go to   adminstration -> jobs and do:
	face detection->Missing   (which does nothing obvious, which feels like a bug)
        then facial recognition->missing

The weird part is, I have my machine learning/facial recognition settings set to:

	model:  buffalo_l
        min detection:  .5
        max recognition:  .4
        min recognized faces:  1

and the facial recognition, on the rerun, will create a LOT of new people - far more than in IMG_1111 picture alone.
So I guess not just this picture is affected, but a bunch of others as well.

Once it's done, then hopefully you have a bunch of faces IDed in that picture.
In practice, you want to touch/update many pictures at once because the Facial recognition process takes a while.

**Comment by bo0tzz** [comment]
This is now supported.

**Reply by 6543** [reply]
is there a pull to link to?

**Reply by WillGunn** [reply]
@6543 most likely https://github.com/immich-app/immich/pull/13784
